DQM/BeamMonitor/plugins/OnlineBeamMonitor.cc

```diff
@@ -0,0 +1,368 @@
+/*
+ * \file OnlineBeamMonitor.cc
+ * \author Lorenzo Uplegger/FNAL
+ * modified by Simone Gennai INFN/Bicocca
+ */
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DQM/BeamMonitor/plugins/OnlineBeamMonitor.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/View.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "RecoVertex/BeamSpotProducer/interface/BeamFitter.h"
+#include "RecoVertex/BeamSpotProducer/interface/PVFitter.h"
+#include <memory>
+
+#include <numeric>
+
+using namespace std;
+using namespace edm;
+using namespace reco;
+
+//----------------------------------------------------------------------------------------------------------------------
+OnlineBeamMonitor::OnlineBeamMonitor(const ParameterSet& ps)
+    : monitorName_(ps.getUntrackedParameter<string>("MonitorName")),
+      bsTransientToken_(
+          esConsumes<BeamSpotObjects, BeamSpotTransientObjectsRcd, edm::Transition::BeginLuminosityBlock>()),
```

Comment:
```suggestion
      bsTransientToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()),
```

Reply:
I have tried to leave as in 11_2_x but it was not compiling so I had to change it to the way as it is now